pkg/controller/v1beta1/inferenceservice/components/explainer.go

```diff
@@ -72,6 +74,14 @@ func (e *Explainer) Reconcile(isvc *v1beta1.InferenceService) (ctrl.Result, erro
 	// StorageInitializer injector to mutate the underlying deployment to provision model data
 	if sourceURI := explainer.GetStorageUri(); sourceURI != nil {
 		annotations[constants.StorageInitializerSourceUriInternalAnnotationKey] = *sourceURI
+		err := isvcutils.ValidateStorageURI(sourceURI, e.client)
+		if err != nil {
+			isvc.Status.UpdateModelTransitionStatus(v1beta1.InvalidSpec, &v1beta1.FailureInfo{
+				Reason:  v1beta1.InvalidStorageURI,
+				Message: "StorageURI is not supported",
+			})
+			return ctrl.Result{}, fmt.Errorf("StorageURI not supported: %v", err)
```

Comment:
model status is only meant for predictor, if the inference service has both explainer and predictor then it can override each other.

Reply:
removed